### ISSUE TITLE
Deprecate `gpu_concat` in favor of `gdf_column_concat`

### ISSUE THREAD

**Issue by jrhemstad** [rapidsai/cudf#526]
## Feature request

https://github.com/rapidsai/cudf/blob/master/cpp/src/compaction/stream_compaction_ops.cu#L387

The old function for concatenating two columns, `gpu_concat`, is wholly superseded by the newer `gdf_column_concat` function. 

Whereas `gpu_concat` only supported concatenating two columns, `gdf_column_concat` supports concatenating an arbitrary number of columns in parallel, making it more general and more efficient.

https://github.com/rapidsai/cudf/blob/branch-0.5/cpp/src/dataframe/column.cpp#L54

`gpu_concat` should therefore be removed in favor using `gdf_column_concat` instead.
